interpreter/subroutine.go

```diff
@@ -20,20 +20,47 @@ func (i *Interpreter) ProcessTestSubroutine(scope context.Scope, sub *ast.Subrou
 	return nil
 }
 
-func (i *Interpreter) ProcessSubroutine(sub *ast.SubroutineDeclaration, ds DebugState) (State, error) {
-	i.process.Flows = append(i.process.Flows, process.NewFlow(i.ctx, sub))
+func (i *Interpreter) subroutineInStack(sub *ast.SubroutineDeclaration) bool {
+	for _, s := range i.Stack {
+		if s.Subroutine == sub {
+			return true
+		}
+	}
+	return false
+}
+
+func (i *Interpreter) pushStackFrame(sub *ast.SubroutineDeclaration) {
+	sf := &StackFrame{
+		Locals:     variable.LocalVariables{},
+		Regex:      make(map[string]*value.String),
+		Subroutine: sub,
+	}
+	i.Stack = append(i.Stack, sf)
```

Comment:
Tentatively we can define the max call stack frames and raise an error if exceeded, probably around 100 would be enough but up to you.

Reply:
Yeah that seems reasonable. I did a test and generated a silly call chain of 200 functions in a fiddle and it didn't cause an error. Seems like the limiting factor isn't the call depth but the overall workspace memory usage which would eventually be exhausted.